Cleanup fix for region being dropped in signup since e6f5ca1

stefanlindbohm opened the pull request:
Cleans up from quick fix in 851756f.

A few learnings:

- Passing a single object to a path helper that doesn't take an ID
  replaces the implicit use of params for that request (in this case
  causing the disappearance of the region param in the path)
- With routes for the same resources defined twice, form_with seems
  confused about which of them to use, even when each action is only
  present in either of the two routes. Specifying both model and url
  solves this.

emmabckstrm reviewed (changes requested):
Nice work! This really cleans everything up as it mostly should be! However, it introduces the signup url having `campaign=` if there is no value for the campaign and I feel that would look unprofessional (I tested it locally). My suggestion is to bring back the 

```
    query_params = {
      lifestyle_footprint: @footprint
    }
    query_params[:campaign] = campaign_param unless campaign_param.blank?
 ```
 
 part, if there is no one liner way of altering as it is now

kallenilver commented:
May I also add a test to the wishlist so that we can make sure this does not happen again?

stefanlindbohm commented:
@emmabckstrm 

> Nice work! This really cleans everything up as it mostly should be! However, it introduces the signup url having `campaign=` if there is no value for the campaign and I feel that would look unprofessional (I tested it locally). My suggestion is to bring back the
> 
> ```
>     query_params = {
>       lifestyle_footprint: @footprint
>     }
>     query_params[:campaign] = campaign_param unless campaign_param.blank?
> ```
> 
> part, if there is no one liner way of altering as it is now

Haha, I thought I did just that but messed up before I made the commit. Fixed now (using `.presence` and no conditionals nor the stuff I did that didn't do anything at all).

@kallenilver 

> May I also add a test to the wishlist so that we can make sure this does not happen again?

Added to the integration test!

emmabckstrm reviewed (approved):
⭐ ⭐ ⭐

kallenilver reviewed (approved):
🥳
